src/js/utils/VersionUtil.js

```diff
@@ -0,0 +1,82 @@
+const VersionUtil = {
+  /**
+   * converts a given versionString to an Object:
+   * {
+   *   major: number,
+   *   minor: number,
+   *   patch: number,
+   *   prerelease: string,
+   *   mnetadata: string
+   * }
+   *
+   * This function mainly tries to return a valid semver-complaint object,
+   * it might truncate invalid input instead of throwing an error.
+   *
+   * @param {string} versionString
+   * @return {object} - an object with versioninformation
+   */
+  convertVersionStringToObject(versionString) {
+    const versionObject = {};
+    let tempVersionArray = [];
+
+    versionObject.metadata = versionString.split("+")[1] || "";
+    versionObject.prerelease = versionString.split("+")[0].split("-")[1] || "";
+
+    tempVersionArray = versionString.split("+")[0].split("-")[0].split(".");
+
+    versionObject.major = parseInt(tempVersionArray[0], 10) || 0;
+    versionObject.minor = parseInt(tempVersionArray[1], 10) || 0;
+    versionObject.patch = parseInt(tempVersionArray[2], 10) || 0;
+
+    return versionObject;
+  },
+  /**
+   * this function compares two given versionstrings.
+   *
+   * both input arguments fall back to an empty string to
+   * prevent this function from throwing and exception
+   *
+   * @param {string} versionStringA - Version A
+   * @param {string} versionStringB - Version B
+   * @return {number} - s-1 if a<b, 0 if a==b, +1 if a>b
+   */
+  compareVersionStrings(versionStringA = "", versionStringB = "") {
+    const verA = this.convertVersionStringToObject(versionStringA);
+    const verB = this.convertVersionStringToObject(versionStringB);
+
+    if (verA.major > verB.major) {
+      return 1;
+    }
+    if (verA.major < verB.major) {
+      return -1;
+    }
+
+    if (verA.minor > verB.minor) {
+      return 1;
+    }
+    if (verA.minor < verB.minor) {
+      return -1;
+    }
+
+    if (verA.patch > verB.patch) {
+      return 1;
+    }
+    if (verA.patch < verB.patch) {
+      return -1;
+    }
+
+    if (!verA.prerelease && verB.prerelease) {
+      return 1;
+    }
+    if (verA.prerelease && !verB.prerelease) {
+      return -1;
+    }
+
+    // cant compare prereleases,
```

Comment:
Since we can't really compare pre-releases, should we block unless they are the same? Like:
```js
if(verA.prerelease !== verB.prerelease) {
  // return whatever blocks the install button
}
```